handler/implementations/procSysCommon.go

```diff
@@ -248,10 +250,14 @@ func (h *ProcSysCommonHandler) Write(
 	// If caching is enabled, store the data in the cache and do a write-through to the
 	// host FS. Otherwise just do the write-through.
 	if h.Cacheable && domain.ProcessNsMatch(process, cntr.InitProc()) {
+
+		cntr.Lock()
```

Comment:
Mmm, for this particular handler this lock could end up acting as a giant-lock if the sys-container is (e.g.) a k8s node, specially if we take into account nsenter's latency. We may need to think about an alternate approach for this in the future. Perhaps we need to partition the /proc/sys hierarchy in various common handlers.

Reply:
Note that it's a per-container lock, so it does not slow down accesses to `proc/sys` from processes in different containers. Let's see if this causes problems from a performance perspective (e.g., if it slows down the k8s tests) and go from there.

Reply:
I understand, that's why i mentioned a 'large' system-container.